### PR TITLE
Update editorConfig

### PR DESCRIPTION
**Pull request by smitpatel**
Did not change the naming convention.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -13,7 +13,7 @@ trim_trailing_whitespace = true
 [*.{cs,csx,vb,vbx}]
 indent_size = 4
 insert_final_newline = true
-#charset = utf-8-bom
+charset = utf-8-bom
```
Comment (AndriySvyryd) [inline .editorconfig]: No BOM please

File: .editorconfig [hunk 2]
```diff
@@ -27,106 +27,148 @@ indent_size = 2
...
 csharp_new_line_before_members_in_object_initializers = true
-csharp_new_line_before_open_brace = all
+csharp_new_line_before_members_in_anonymous_types = true
 csharp_new_line_between_query_expression_clauses = true
 
 # Identation options
+csharp_indent_block_contents = true
+csharp_indent_braces = false
+csharp_indent_case_contents_when_block = true
+csharp_indent_switch_labels = true
```
Comment (AndriySvyryd) [inline .editorconfig]: repeated

File: src/EFCore.Cosmos/Query/Pipeline/SelectExpression.cs [hunk 3]
```diff
@@ -13,17 +13,17 @@ namespace Microsoft.EntityFrameworkCore.Cosmos.Query.Pipeline
 {
     public class SelectExpression : Expression
     {
-        private const string _rootAlias = "c";
+        private const string RootAlias = "c";
```
Comment (AndriySvyryd) [inline src/EFCore.Cosmos/Query/Pipeline/SelectExpression.cs]: We've decided at some point that this should use the underscore case

File: src/EFCore.Cosmos/Update/Internal/DocumentSource.cs [hunk 4]
```diff
@@ -1,4 +1,4 @@
-// Copyright (c) .NET Foundation. All rights reserved.
+﻿// Copyright (c) .NET Foundation. All rights reserved.
```
Comment (AndriySvyryd) [inline src/EFCore.Cosmos/Update/Internal/DocumentSource.cs]: Revert encoding

File: src/EFCore.Cosmos/Query/Pipeline/SelectExpression.cs (hunk 3 above)
Comment (smitpatel) [reply src/EFCore.Cosmos/Query/Pipeline/SelectExpression.cs]: Adding rule to make private const fields to _camelCase

**Pull request merged**